black-mulberry-core/src/main/java/com/black/mulberry/core/service/ProductService.java

```diff
@@ -0,0 +1,113 @@
+package com.black.mulberry.core.service;
+
+import com.black.mulberry.core.entity.CategoryProduct;
+import com.black.mulberry.core.entity.Product;
+import com.black.mulberry.core.entity.User;
+import com.black.mulberry.core.exception.ProductNotExistException;
+import com.black.mulberry.core.mapper.ProductMapper;
+import com.black.mulberry.core.repository.CategoryProductRepository;
+import com.black.mulberry.core.repository.ProductRepository;
+import com.black.mulberry.core.repository.UserRepository;
+import com.black.mulberry.data.transfer.request.ProductRequest;
+import com.black.mulberry.data.transfer.response.ProductResponse;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.io.IOUtils;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.persistence.EntityManager;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+@Service
+@RequiredArgsConstructor
```

Comment:
have a separate interface ProductService and move the business logic of the ProductServiceImp package into Impl